### DISCUSSION TITLE
Kubernetes ♻️ Practices & Scheduling

### DISCUSSION THREAD

**Discussion by mkorbi** [cncf/tag-env-sustainability#23] [Project Proposal]
### Description
Kubernetes is the key driver and foundation of the cloud native universe, within this project we will work on the basic capabilities, best practices and scheduling mechanism as well as extensions to improve the K8s GHG impact.

### Impact
Where are patterns there are anti-patterns, where we improve for speed, resistance and security there is space for configurations and setting with a negative impact on power consumption and the following chain.

Cluster sprawls, container sprawls, false scheduling, wrong resource limits/requests, no descheduling all these are very default capabilities of Kubernetes but most of them are not done at all or are very wrong, treating K8s like a VM system from the early 2000's.

### Scope
We should try in the beginning to stay with the very basic capabilities of Kubernetes as well as anything that is in the context of scheduling.

**Comment by tomkerkhove** [comment]
Happy to help from a [KEDA](https://keda.sh) side.

**Reply by mkorbi** [reply]
Keda would be awesome, I think we also some of our friends from Mia Platform and [kube-green](https://kube-green.dev/) in the group, and I think maybe [karpenter](https://karpenter.sh/) could be also in interest

**Comment by catblade** [comment]
Request that these items are also sent to the mailing list, where discussions historically happen.

I will comment that there are various cloud native projects and there are other technologies that should not be discounted, such as WASM, and focusing our efforts solely on one part of the ecosystem does not future-proof our tech.

Please spell out things like GHG.  Also, I'd request we agree on terminology within this space, as "carbon" is the more common term in the industry at this point.

I would improve the wording of the impact, which I find unfair because the default was built for cases other than ours, to:

Kubernetes scheduling was not designed for complex workloads or carbon-aware scheduling.  It does, however, allow for extensions or plugins within the scheduling space.  We should be looking at projects that improve this space, and there are many.

**Reply by mkorbi** [reply]
There is a couple of points in it.

> Request that these items are also sent to the mailing list, where discussions historically happen.

I wouldn't discuss anything on a mailing list, it is impossible to follow a discussion and it is spam. Leading the people through a mail to the discussions is good.

> I will comment that there are various cloud native projects and there are other technologies that should not be discounted, such as WASM, and focusing our efforts solely on one part of the ecosystem does not future-proof our tech.

It is too much if we want to start with everything at the same it will be impossible to deliver anything. And at such tech there are communities in which we can talk to rather than invent the things again.

**Comment by leonardpahlke** [comment]
I think it's a good idea to start with Kubernetes as this is the foundation for most cloud native systems and therefore discoveries made here have potentially a very large impact on advancing environmental sustainability in the cloud native space.

I'm happy to actively help move this particular project forward 🙌.

**Comment by mkorbi** [comment]
https://github.com/cncf/tag-env-sustainability/issues/54

**Comment by catblade** [comment]
One of the points of the landscape document is to cover projects such as this.  There are a variety of project that cover this in different ways.

Can we go through these projects and close the discussions that we are covering in the landscape doc?  Re-open others later?
